components/feature/tabs/src/main/java/org/mozilla/rocket/tabs/SessionManager.kt

```diff
@@ -82,6 +82,7 @@ class SessionManager @JvmOverloads constructor(
         for (state in states) {
             if (state.session.isValid()) {
                 getOrCreateEngineSession(state.session).let { link(state.session, it) }
+                state.session.engineSession!!.webViewState = state.engineSession?.webViewState
```

Comment:
state.session.engineSession could be null.
Now we have some !! in the codebase and we lost Kotlin's null-safety benefit. Maybe we should think of a better way to fix it.

Reply:
Yeah I thought it's impossible but I'm wrong, thanks!